### PR TITLE
week2 ff

### PR DESCRIPTION
**Pull request by hieuvo**

### CODE REVIEW

File: week0002/hieuvo/week0002.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+foreach ($path as $direction) {
+    list($x, $y) = move($x, $y, $direction);
+    $spiral[$x][$y] = $v++;
+}
+
+$stdout = fopen('php://stdout', 'w');
+
+$i = 1;
+foreach ($spiral as $row) {
+    fwrite($stdout, implode(' ', $row));
```
Comment (vietlq) [inline week0002/hieuvo/week0002.php]: Please test against the sample file provided.

File: week0002/hieuvo/week0002.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    if ($i++ < $M) { 
+        fwrite($stdout, "\n");
```
Comment (vietlq) [inline week0002/hieuvo/week0002.php]: Please test against the sample file provided.

Comment (hieuvo) [inline week0002/hieuvo/week0002.php]: hmm, I'm aware of this issue, but in your problem statement, you didn't mention about indent between numbers when printing them out, btw I'll change

File: week0002/hieuvo/week0002.php [hunk 3]
```diff
@@ -2,6 +2,7 @@
```
Comment (hieuvo) [inline week0002/hieuvo/week0002.php]: @vietlq : here is the fix

File: week0002/hieuvo/week0002.php (hunk 2 above)
Comment (vietlq) [inline week0002/hieuvo/week0002.php]: I requested to test your output against the file `test_case_01.dat`. That should be enough to cover spacing and new lines. Please make testing a new habit!

File: week0002/hieuvo/week0002.php [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 $n = 4; // columns
 $m = $M = 4; // rows
+$length = ceil(log10(abs($n * $m) + 1));
```
Comment (vietlq) [inline week0002/hieuvo/week0002.php]: This is good.

**Pull request merged**